src/main/java/eu/dissco/orchestration/backend/service/SourceSystemService.java

```diff
@@ -17,55 +19,139 @@
 import eu.dissco.orchestration.backend.exception.PidAuthenticationException;
 import eu.dissco.orchestration.backend.exception.PidCreationException;
 import eu.dissco.orchestration.backend.exception.ProcessingFailedException;
+import eu.dissco.orchestration.backend.properties.TranslatorJobProperties;
 import eu.dissco.orchestration.backend.repository.SourceSystemRepository;
 import eu.dissco.orchestration.backend.web.HandleComponent;
+import freemarker.template.Configuration;
+import freemarker.template.TemplateException;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.BatchV1Api;
+import io.kubernetes.client.openapi.models.V1Container;
+import io.kubernetes.client.openapi.models.V1CronJob;
+import io.kubernetes.client.openapi.models.V1EnvVar;
+import io.kubernetes.client.openapi.models.V1Job;
+import java.io.IOException;
+import java.io.StringWriter;
 import java.time.Instant;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class SourceSystemService {
+
   public static final String SUBJECT_TYPE = "SourceSystem";
   private final FdoRecordService fdoRecordService;
   private final HandleComponent handleComponent;
   private final SourceSystemRepository repository;
   private final MappingService mappingService;
   private final KafkaPublisherService kafkaPublisherService;
   private final ObjectMapper mapper;
+  @Qualifier("yamlMapper")
+  private final ObjectMapper yamlMapper;
+  private final TranslatorJobProperties jobProperties;
+  private final Configuration configuration;
+  private final BatchV1Api batchV1Api;
 
-  public JsonApiWrapper createSourceSystem(SourceSystem sourceSystem, String userId, String path)
-      throws NotFoundException {
-    String handle = null;
-    var request = fdoRecordService.buildCreateRequest(sourceSystem, ObjectType.SOURCE_SYSTEM);
-    try {
-      handle = handleComponent.postHandle(request);
-    } catch (PidAuthenticationException | PidCreationException e) {
-      throw new ProcessingFailedException(e.getMessage(), e);
+  private static String getSuffix(String sourceSystemId) {
+    return sourceSystemId.substring(sourceSystemId.indexOf('/') + 1).toLowerCase();
+  }
+
+  private static String generateJobName(SourceSystemRecord sourceSystem, boolean isCron) {
+    var name = sourceSystem.sourceSystem().translatorType().name().toLowerCase() + "-" + getSuffix(
```

Comment:
The name function is available by default for enums. It just returns the name of the enum as a string. So in this case it will be the same as the `literal`. Changed it to the literal, as that might be better understandable.

Reply:
didn't know that about enums! Either function should be good